endpoints/version.go

```diff
@@ -7,31 +7,33 @@ import (
 	"github.com/golang/glog"
 )
 
-type versionModel struct {
-	Version  string `json:"version"`
-	Revision string `json:"revision"`
-}
+const versionNotSet = "not-set"
 
 // NewVersionEndpoint returns the latest git tag as the version and commit hash as the revision from which the binary was built
-func NewVersionEndpoint(version string, revision string) http.HandlerFunc {
-	if version == "" {
-		version = "not-set"
-	}
-	if revision == "" {
-		revision = "not-set"
+func NewVersionEndpoint(version, revision string) http.HandlerFunc {
+	response, err := prepareVersionResponse(version, revision)
+	if err != nil {
+		glog.Fatalf("error creating /version endpoint response: %v", err)
 	}
 
 	return func(w http.ResponseWriter, _ *http.Request) {
-		jsonOutput, err := json.Marshal(versionModel{
-			Version:  version,
-			Revision: revision,
-		})
-		if err != nil {
-			glog.Errorf("/version Critical error when trying to marshal versionModel: %v", err)
-			w.WriteHeader(http.StatusInternalServerError)
-			return
-		}
+		w.Write(response)
+	}
+}
 
-		w.Write(jsonOutput)
+func prepareVersionResponse(version, revision string) (json.RawMessage, error) {
+	if version == "" {
+		version = versionNotSet
 	}
+	if revision == "" {
+		revision = versionNotSet
+	}
+
+	return json.Marshal(struct {
+		Revision string `json:"revision"`
+		Version  string `json:"version"`
+	}{
+		Revision: revision,
+		Version:  version,
+	})
 }
```

Comment:
Updated the version endpoint to create a static response once at startup, mimicking the pattern used for the bidder info handler. Changed response structure to list the revision first to match PBS-Java. This is not a breaking change for the admin endpoint and we might as well make it match for the public debut.